python/src/spark_rapids_ml/knn.py

```diff
@@ -1508,7 +1514,27 @@ def _transform_internal(
                 if isinstance(item, np.ndarray):
                     item = cp.array(item, dtype="float32")
 
-                index_obj = nn_object.build(build_params, item)
+                try:
+                    index_obj = nn_object.build(build_params, item)
+                except Exception as e:
+                    if "k must be less than topk::kMaxCapacity (256)" in str(e):
```

Comment:
Interesting way to do it.  What is advantage over  checking the error condition before invoking nn_object.build?

Reply:
topk::kMaxCapacity (currently 256) is a global variable which may be updated to a higher or lower value some day. Hopefully we can catch the potential update by checking the error message after nn_object.build.